### PR TITLE
Broke tests apart and added fixtures

### LINKED ISSUE #56: Use separate, simple fixture sites for tests
Right now, the `build` test runs against `examples/simple-site`, which includes testing jade, coffeescript and stylus templates all in one test.

First, that makes our tests rely on the example site, so if we change the example site for documentation / quick-start purposes, it may break our tests.

Second, it lumps all building tests into a single build. Does it make sense to have one fixture site to test jade, one to test coffeescript, one to test stylus etc? This doesn't seem like a huge deal right now since those three things cover separate concerns, but if we add a markdown test, and start adding additional asset types, the single simple site will get cluttered and hard to manage.

### PR DESCRIPTION
**Pull request by diminish7**
closes #56 

1. Broke the main `build` test up into tests for individual asset types
2. Added fixtures directory to house small apps for testing individual bits of functionality, rather than adding more and more to the main example app
3. Added some helpers:
   * Build helpers to run some common functions between each of the asset build tests
   * Runners to run the `build` and `serve` tasks

**Note** that the `build` and `serve` runner helpers are placed in their own file so that they can be launched in their own forked process. The problem with running them inline is that they need to be run from each fixture app's root directory, but their async nature makes it impossible to know which dir you're in when the task actually fires. By running them in their own process, we can `chdir` to the right directory, then fork the process, which stays in that directory no matter where the main process `chdir`s to next.

### CODE REVIEW

File: test/xml_spec.coffee [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{buildAndVerify} = require './helpers/build_helpers'
+
+assert = require "assert"
+{exists} = require "fairmont"
```
Comment (f1337) [inline test/xml_spec.coffee]: I don't see any use of `assert` or `exists` in this file.

File: test/stylus_spec.coffee [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{buildAndVerify} = require './helpers/build_helpers'
+
+assert = require "assert"
+{exists} = require "fairmont"
```
Comment (f1337) [inline test/stylus_spec.coffee]: I don't see any use of `assert` or `exists` in this file.

File: test/json_spec.coffee [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{buildAndVerify} = require './helpers/build_helpers'
+
+assert = require "assert"
+{exists} = require "fairmont"
```
Comment (f1337) [inline test/json_spec.coffee]: I don't see any use of `assert` or `exists` in this file.

File: test/jade_spec.coffee [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{buildAndVerify} = require './helpers/build_helpers'
+
+assert = require "assert"
+{exists} = require "fairmont"
```
Comment (f1337) [inline test/jade_spec.coffee]: I don't see any use of `assert` or `exists` in this file.

File: test/coffeescript_spec.coffee [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{buildAndVerify} = require './helpers/build_helpers'
+
+assert = require "assert"
+{exists} = require "fairmont"
```
Comment (f1337) [inline test/coffeescript_spec.coffee]: I don't see any use of `assert` or `exists` in this file.

File: test/serve_spec.coffee [hunk 6]
```diff
@@ -1,17 +1,26 @@
+{build, isBuilt} = require "./helpers/build_helpers"
+
 {join} = require "path"
 {run} = require "panda-9000"
```
Comment (f1337) [inline test/serve_spec.coffee]: I don't see `run` used.

File: test/serve_spec.coffee [hunk 7]
```diff
@@ -1,17 +1,26 @@
...
 assert = require "assert"
-{shell, sleep, call, exists, promise} = require "fairmont"
+{shell, sleep, call, exists, chdir, promise} = require "fairmont"
```
Comment (f1337) [inline test/serve_spec.coffee]: I only see `chdir` and `promise` in use in this file

**Comment by f1337** [comment]
While I gave my endorsement in the issue conversation (https://github.com/pandastrike/haiku9/issues/56#issuecomment-234040470), I'm now wondering whether the complexity introduced by building multiple fixture sites is worth the tradeoff?

File: test/serve_spec.coffee (hunk 7 above)
Comment (diminish7) [inline test/serve_spec.coffee]: Yep, good call on all these unused `require`s. Leftover cruft from refactoring. I'll kill em all.

**Comment by diminish7** [comment]
@f1337 Yeah, I hear that, but the complexity is largely encapsulated now, so adding new tests with this pattern is easy, and it keeps each of us more segregated in our work so we're less likely to step on each others' toes. Plus it makes the tests easier to reason about. When I run the `jade` test, I expect to see a `jade` template compiled.

So yeah, I think it's worth the tradeoff.

**Comment by f1337** [comment]
:+1:

**Pull request merged**